### ISSUE TITLE
one sensible reason to store as much data?

### ISSUE THREAD

**Issue by gUseg** [lucko/LuckPerms#640]
what a stupid architecture in the database of this plugin? why so much is required cached data like EVERY JOINED PLAYER uuids and his DEFAULT group in other table?
why inheritance can not be assigned in the same TABLE? why for the prefixes and suffixes can not be allocated personal table?

**Comment by BrainStone** [comment]
The way the data is stored is in fact very sensical. It stored in a fashion referred to as “normalized”. The upside of this way to store data is that it eliminates duplicate data as much as possible.

Now to answer your question why the group isn’t stored in the player table. That’s very simple. You need a table for player permissions anyways. So why not store the parent groups there. Also if you were to store the parent groups in the main table you couldn’t allow multiple inheritance.

Next, I’m not quite sure what you mean by personal table, but if you mean a table per player for their prefixes a d suffixes, that would be plain stupid. It would defy any database design principle. It makes sense storing them in the player and group permissions tables.

You also complain about every joined player being stored. Why exactly is that an issue? It simplifies a fair amount of operations. And barely costs any data whatsoever. I mean we’re talking about less than 100 bytes per player. So even a million players wouldn’t even need 100 MB.

To summarize the database has a very logical structure and follows the current recommendations for databases. Your question seems naive and like you are not very familiar with databases. So while you might not quite understand why it makes sense the way it is done, I can tell you it is set up in one of the best ways possible.

**Comment by lucko** [comment]
Mindless hate and criticism is not constructive and is unappreciated here.

If you have positive suggestions and are willing to contribute them in a friendly way, please open a new issue.

Thank you.
